LogicBuilder.Expressions.Utils/TypeExtensions.cs

```diff
@@ -204,12 +204,21 @@ private static MemberInfo[] GetValueTypeMembers(this Type parentType)
             return parentType.GetMemberInfos().Where
             (
                 info => (info.MemberType == MemberTypes.Field || info.MemberType == MemberTypes.Property)
-                && (info.GetMemberType().IsLiteralType() || info.GetMemberType().IsLiteralList())
+                && info.GetMemberType().AnyLiteralTypes()
```

Comment:
1) Can you say which [PostgreSQL](https://www.npgsql.org/doc/types/basic.html) types we would be supporting?  Your test will pass for `KeyValuePair<int, string>` not just lists and dictionaries.

2) It's also easier to reverse changes if the update is something like:
```c#
            return parentType.GetMemberInfos().Where
            (
                info => (info.MemberType == MemberTypes.Field || info.MemberType == MemberTypes.Property)
                && (info.GetMemberType().IsLiteralType() || info.GetMemberType().IsLiteralList() || info.GetMemberType().AllUnderlyingTypesLiteral())
            ).ToArray();
```
3) You're returning the type itself from `UnderlyingElementTypes()` in place of the call to `IsLiteralType()` - also harder to reverse.

Reply:
1. I think it is up to the developer to decide which structure their DB supports, in my case, since OData doesn't natively support $filter on dictionaries, I want to use [EF Value Conversation](https://learn.microsoft.com/en-us/ef/core/modeling/value-conversions?tabs=data-annotations#pre-defined-conversions) to treat dictionaries as List<KeyValuePair<,>,>

2. // 3. Not sure I understand your concern, is it about rollback the changes in case of a regression/bug? If so, can't we create a rollback PR to the current PR?.
 In addition, since AllUnderlyingTypesLiteral() is also covering Lists I don't see a point keeping IsLiteralList()

Reply:
3.) note that I'm calling . AllUnderlyingTypesLiteral() which returns a boolean, not the type itself (it used .All that calls UnderlyingElementTypes)

Reply:
What you've written is `GenericTypeOne<int, string, int, GenericTypeTwo<string, int, int, string>>` which is any object graph.

Sounds like you need GenericTypeOne to be an IEnumerable with the with `arg[0]` as a literal type? What generic arguments does the PastgreSQL dictionary allow?

I wasn't referring to operationally reverting a commit.  Keeping those 3 groups of literal types separate makes it easier to address/see/understand a problem with any one of them.  `AnyLiteralTypes` does the reverse.

Reply:
Ideally, we want to expand all non-navigation properties by default.  Literal lists and literals etc.. approximate that behavior but leave out a few properties.  It's probably better to copy the expansion logic to AutoMapper.AspNetCore.OData and use OData's metadata to identify non-navigation properties.  `IEdmEntityType` has a `NavigationProperties()` extension method.   Here's a [similar use of metadata](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/05a00689d8c43ed30f9561dea24cbc6f8ac40868/AutoMapper.AspNetCore.OData.EFCore/ODataQueryContextExtentions.cs#L29).

Agree?

Reply:
Should the logic be in [QueryableExtensions](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/master/AutoMapper.AspNetCore.OData.EF6/QueryableExtensions.cs#L188)?